### ISSUE TITLE
Extern declarations

### ISSUE THREAD

**Issue by litherum** [gpuweb/gpuweb#883]
If the group doesn't accept https://github.com/gpuweb/gpuweb/issues/875 nor https://github.com/gpuweb/gpuweb/issues/741, we can at least make it _possible_ for authors to write source that doesn't depend on include orderings. If we allow extern declarations for functions and global variables and global types, each source file can forward-declare everything it uses from a different file.

**Comment by kainino0x** [comment]
I support something like this, along with linking (real linking, not shader concatenation), but not until after MVP.

**Comment by grorg** [comment]
Discussed at the [2020-08-18 meeting](https://docs.google.com/document/d/1P7XNQ760P5yONHTkhGAlP-AA2VpggNDGMGHaGAE2tn0/edit#).

Resolution was that this is not needed for MVP, unless @litherum has some use cases.

**Comment by litherum** [comment]
We have out-of-order globals. Closing.
